src/darwin/Framework/CHIP/MTRDeviceController_Concrete.mm

```diff
@@ -1619,6 +1619,17 @@ - (void)invalidateCASESessionForNode:(NSNumber *)nodeID;
     [self syncRunOnWorkQueue:block error:nil];
 }
 
+- (void)invalidateCASESessionEstablishmentForNode:(NSNumber *)nodeID;
+{
+    auto block = ^{
+        auto caseSessionMgr = self->_cppCommissioner->CASESessionMgr();
+        VerifyOrDie(caseSessionMgr != nullptr);
+        caseSessionMgr->ReleaseSession(self->_cppCommissioner->GetPeerScopedId(nodeID.unsignedLongLongValue));
+    };
+
+    [self syncRunOnWorkQueue:block error:nil];
```

Comment:
Why do we need a sync dispatch here?

Reply:
https://github.com/project-chip/connectedhomeip/pull/36315